### PR TITLE
 Merging Max's API syntax changes in.

### PR DESCRIPTION
**Pull request by dkorolev**
Hi @mzhurovich,

It's a bit of work in progress, but your https://github.com/mzhurovich/Sherlock/commit/a484a39ce593eb235be6683ae6853d77c71afcb4 is now in. I took the liberty to comment out a few methods -- we'll deprecate or re-add them soon.

Thanks,
Dima

### CODE REVIEW

**Comment by dkorolev** [comment]
@mzhurovich : PTAL :-)

File: yoda/api/key_entry/key_entry.h [hunk 1]
```diff
@@ -211,31 +211,87 @@ struct Container<YT, KeyEntry<ENTRY>> {
...
     }
+
+   private:
+    Container<YT, YET>& mutable_;
+    typename YT::T_STREAM_TYPE& stream_;
+  };
+
+  Accessor operator()(container_wrapper::RetrieveAccessor<YET>) const { return Accessor(*this); }
+
+  // TODO(dkorolev): This one should not be const.
```
Comment (mzhurovich) [inline yoda/api/key_entry/key_entry.h]: Looks like it's not const :)

Comment (dkorolev) [inline yoda/api/key_entry/key_entry.h]: Fixed!

File: yoda/test.cc [hunk 2]
```diff
@@ -61,55 +61,80 @@ TEST(Yoda, CoverTest) {
...
+    // `Get()` syntax.
+    EXPECT_EQ(42.0, static_cast<const KeyValueEntry&>(KVEntries.Get(1)).value);
+    EXPECT_FALSE(KVEntries.Get(56));
+
+    // `operator[]` syntax.
+    EXPECT_EQ(42.0, static_cast<const KeyValueEntry&>(KVEntries[1]).value);
+    KeyValueEntry kve34;
+    ASSERT_THROW(kve34 = KVEntries[34], yoda::KeyNotFoundCoverException);
+
+    auto mutable_kve = cw.GetMutator<KeyEntry<KeyValueEntry>>();
```
Comment (mzhurovich) [inline yoda/test.cc:79]: +1 for this syntax!

**Comment by mzhurovich** [comment]
@dkorolev Looks good to go further!

**Comment by dkorolev** [comment]
Yay!

**Pull request merged**